metadata-models/src/main/pegasus/com/linkedin/common/FieldTransformationType.pdl

```diff
@@ -0,0 +1,21 @@
+namespace com.linkedin.common
+
+/**
+ * Type of the transformation involved in generating destination fields from source fields.
+ */
+enum FieldTransformationType{
+  /**
+   * BlackBox transformation doesn't know what transformations are involved in producing the target field from source fields.
+   */
+  BLACKBOX,
+
+  /**
+   * There is transformation involved from source to destiantion fields and producers of the metadata can specify how that transformation happened.
+   */
+  CUSTOM
```

Comment:
I would rename this as UDF since you are allowing user to only specify a transformation function.